spec/brainstem/api_docs/formatters/open_api_specification/version_2/endpoint/response_definitions_formatter_spec.rb

```diff
@@ -0,0 +1,335 @@
+                stub(endpoint).presenter { presenter }
+                stub(endpoint).action { action }
+                stub(endpoint).custom_response_configuration_tree { {} }
+                stub(endpoint).http_methods { http_methods }
+              end
+
+              describe '#call' do
+                context 'when delete request' do
+                  let(:http_methods) { %w(DELETE) }
+
+                  it 'formats the delete response and error response' do
+                    any_instance_of(described_class) do |instance|
+                      mock(instance).format_delete_response!
+                      mock(instance).format_error_responses!
+
+                      dont_allow(instance).format_schema_response!
+                    end
+
+                    subject.call
+                  end
+                end
+
+                context 'when request is not delete' do
+                  let(:http_methods) { %w(GET) }
+
+                  it 'formats the schema response and error response' do
+                    any_instance_of(described_class) do |instance|
+                      mock(instance).format_schema_response!
+                      mock(instance).format_error_responses!
+
+                      dont_allow(instance).format_delete_response!
+                    end
+
+                    subject.call
+                  end
+                end
+
+              end
+
+              describe '#formatting' do
+                describe '#success_response_description' do
+                  subject { described_class.new(endpoint).send(:success_response_description) }
+
+                  context 'when `GET` request' do
+                    let(:http_methods) { %w(GET) }
+
+                    it { is_expected.to eq('A list of Widgets have been retrieved.') }
+                  end
+
+                  context 'when `POST` request' do
+                    let(:http_methods) { %w(POST) }
+
+                    it { is_expected.to eq('Widget has been created.') }
+                  end
+
+                  context 'when `PUT` request' do
+                    let(:http_methods) { %w(PUT) }
+
+                    it { is_expected.to eq('Widget has been updated.') }
+                  end
+
+                  context 'when `PATCH` request' do
+                    let(:http_methods) { %w(PATCH) }
+
+                    it { is_expected.to eq('Widget has been updated.') }
+                  end
+
+                  context 'when `DELETE` request' do
+                    let(:http_methods) { %w(DELETE) }
+
+                    it { is_expected.to eq('Widget has been deleted.') }
+                  end
+                end
+
+                describe '#format_delete_response!' do
+                  let(:http_methods) { %w(DELETE) }
+
+                  it 'returns the structure response for a destroy action' do
+                    subject.send(:format_delete_response!)
```

Comment:
Are we `send`-ing a private method here? Presumably that's easier, and is there no higher-level public method we could call?

Reply:
Yes. I generally shy away from explicitly testing private methods but I decided to follow the pattern used in testing formatters in the project. It definitely makes it easier & more readable to test these private methods otherwise I would have to stub out a bunch of format methods to just test this one thing